### PR TITLE
Inset shadows

### PR DESCRIPTION
**Pull request by Vidhanvyrs**
changes done on props.shadows.js according to the codepen example that was given with the issue
Please provide feedback and point out if any particular changes required

### CODE REVIEW

**Review by argyleink — CHANGES_REQUESTED** [review]
thanks for working on this! ❤️ 

can you reduce the PR changes to just the code added please? looks like a formatter ran and is making it hard to decipher what you added.

can you change the shadow additions to only be the bottom inset color, like this `inset 0 -.5px 0 0 #fff1`?

remove the modifications to `--inner-shadow-0` it doesnt need the trick

run `npm run lib:shadows` to generate the css file from the changes to this js file. 

After these changes, I'll create a few tester examples to make sure the trick isnt disrupting anything and is in fact an improvement. Thanks again! no one will probably notice this change, but one day, they'll see its just slightly more realistic 😁

**Comment by Vidhanvyrs** [comment]
Thankyou @argyleink 
The required  changes will be done in no time 
I'M ON IT!

**Comment by Vidhanvyrs** [comment]
Extremely sorry i accidently added the commit without running the npm run lib: shadows in the commit [7e9aa4c](https://github.com/argyleink/open-props/pull/393/commits/7e9aa4cb43536f58309fec78b531507ccf8dce55)
but i do performed it in commit [0d1ba17](https://github.com/argyleink/open-props/pull/393/commits/0d1ba179f076cd86db0e595b7a0b0a864f81728b)
So now i am clarifying  about the changes! Please forgive 🙇‍♂️

File: src/props.shadows.js [hunk 1]
```diff
@@ -1,35 +1,36 @@
...
-  '--shadow-strength-@media:dark': '25%',
-  
-  '--shadow-1': '0 1px 2px -1px hsl(var(--shadow-color) / calc(var(--shadow-strength) + 9%))',
-  '--shadow-2': `
+  "--shadow-color-@media:dark": "220 40% 2%",
+  "--shadow-strength-@media:dark": "25%",
+
+  "--shadow-1":
+    "0 1px 2px -1px hsl(var(--shadow-color) / calc(var(--shadow-strength) + 9%))",
+  "--shadow-2": `
```
Comment (argyleink) [inline src/props.shadows.js:10]: all of this is unchanged right? it's just the formatter changed the quotes?

File: src/props.shadows.js [hunk 2]
```diff
@@ -38,23 +39,33 @@ const Shadows = {
     0 41px 33px -2px hsl(var(--shadow-color) / calc(var(--shadow-strength) + 6%)),
     0 100px 80px -2px hsl(var(--shadow-color) / calc(var(--shadow-strength) + 7%))`,
 
-  '--inner-shadow-0': 'inset 0 0 0 1px hsl(var(--shadow-color) / calc(var(--shadow-strength) + 9%))',
-  '--inner-shadow-1': 'inset 0 1px 2px 0 hsl(var(--shadow-color) / calc(var(--shadow-strength) + 9%))',
-  '--inner-shadow-2': 'inset 0 1px 4px 0 hsl(var(--shadow-color) / calc(var(--shadow-strength) + 9%))',
-  '--inner-shadow-3': 'inset 0 2px 8px 0 hsl(var(--shadow-color) / calc(var(--shadow-strength) + 9%))',
-  '--inner-shadow-4': 'inset 0 2px 14px 0 hsl(var(--shadow-color) / calc(var(--shadow-strength) + 9%))',
-}
+  "--inner-shadow-0":
```
Comment (argyleink) [inline src/props.shadows.js:42]: inner-shadow-0 shouldnt have any changes 🙂

File: src/props.shadows.js [hunk 3]
```diff
@@ -38,23 +39,33 @@ const Shadows = {
...
+    "inset 0 0 0 1px hsl(var(--shadow-color) / calc(var(--shadow-strength) + 9%))",
+
+  "--inner-shadow-1": `
+    inset 0 -.5px 0 0 #fff1`,
+  "--inner-shadow-2": `
+    inset 0 -.5px 0 0 #fff1`,
+  "--inner-shadow-3": `
+    inset 0 -.5px 0 0 #fff1`,
+  "--inner-shadow-4": `
+    inset 0 -.5px 0 0 #fff1`,
```
Comment (argyleink) [inline src/props.shadows.js:52]: oh no, looks like the inner shadows are gone and now there's only the highlight. typo maybe?!

File: src/props.shadows.js [hunk 4]
```diff
@@ -38,23 +39,33 @@ const Shadows = {
...
+        Shadows["--shadow-strength"]
+      );
+      return [key, value];
     })
-    .filter(([key]) =>
-      !['--shadow-color', '--shadow-strength'].includes(key)
-    )
-)
+    .filter(([key]) => !["--shadow-color", "--shadow-strength"].includes(key))
+);
```
Comment (argyleink) [inline src/props.shadows.js:71]: formatter changes again? this should be unchanged

**Review by argyleink — CHANGES_REQUESTED** [review]

File: src/props.shadows.js (hunk 3 above)
Comment (Vidhanvyrs) [reply src/props.shadows.js:52]: oops!
 that's a typo

File: src/props.shadows.js (hunk 1 above)
Comment (Vidhanvyrs) [reply src/props.shadows.js:10]: yes it's the formatter I have just changed the inner shadows 
and that too i performed a typo by putting fff1 instead hsl

File: src/props.shadows.js (hunk 3 above)
Comment (Vidhanvyrs) [reply src/props.shadows.js:52]: yeah that is a typo sorry correcting it

File: src/props.shadows.js (hunk 4 above)
Comment (Vidhanvyrs) [reply src/props.shadows.js:71]: how can i do this no formatter interference because before committing, the code is the same before but when i committed its the formatter that is doing this 
Can you please suggest me how can i resolve this formatter thing this is what chatgpt suggested 

If you want to prevent your code formatter from automatically formatting your code upon committing, you can take a few steps depending on the tools you're using. Here's how you can do it using Git and popular code formatters like Prettier:

Git Hooks: You can use Git hooks to run scripts before or after certain Git events, such as committing. You can create a pre-commit hook that temporarily disables the formatter before committing. Here's a general idea:

Create a .pre-commit-config.yaml file in your project's root directory (if it doesn't already exist).

Configure the hook to disable the formatter for the staged files using the formatter's command-line options. For example, if you're using Prettier, you can run:

yaml
Copy code
-   repo: https://github.com/pre-commit/mirrors-prettier
    rev: v2.5.1
    hooks:
    - id: prettier
      args: ["--write", "$PRE_COMMIT_HOME/../$PRE_COMMIT_FILES"]
This configuration will use Prettier to format the files, but it will only format the staged files as specified by $PRE_COMMIT_FILES.

Commit the .pre-commit-config.yaml file along with your code.

Configuration Files: Some code formatters, like Prettier, allow you to configure their behavior through configuration files like .prettierrc or prettier.config.js. You can include these configuration files in your project and customize the formatting options to match your preferred style. This way, the formatter will consistently apply the same formatting rules.

Remember, while these methods can help you maintain a consistent code style, it's also a good practice to collaborate with your team and agree on a shared code formatting style that everyone follows. This can help reduce conflicts related to formatting in pull requests.

Comment (argyleink) [reply src/props.shadows.js:71]: i think you need to disable your formatter in this project from running when you press save. it's a common feature people have when they work on teams and they've agreed on that formatter, but this project is different and your formatter is trying to do the right thing based on it's settings, but it's also making your PR change more than it needs to. i dont know which formatter you have installed on your machine, so i can't give particular instructions. but you could try disabling it entirely while editing this file and then enabling it again once back on your team?

sorry this sounds tedious! but it's also good practice to make the smallest change needed for the feature. changing a whole files single quotes to double quotes, is more than what's needed for the feature. thanks for working some of those formatter changes back, sorry they're in your way.

i'm also on vacation =) will check back here in a couple days

File: src/props.shadows.js (hunk 1 above)
Comment (argyleink) [inline src/props.shadows.js:10]: these changes shouldn't be part of the PR 😅

File: src/props.shadows.js [hunk 5]
```diff
@@ -38,23 +39,32 @@ const Shadows = {
...
+    "inset 0 0 0 1px hsl(var(--shadow-color) / calc(var(--shadow-strength) + 9%))",
+  "--inner-shadow-1": `
+    inset 0 -.5px 0 0 hsl(var(--shadow-color) / calc(var(--shadow-strength) + 9%))`,
+  "--inner-shadow-2": `
+    inset 0 -.5px 0 0 hsl(var(--shadow-color) / calc(var(--shadow-strength) + 9%))`,
+  "--inner-shadow-3": `
+    inset 0 -.5px 0 0 hsl(var(--shadow-color) / calc(var(--shadow-strength) + 9%))`,
+  "--inner-shadow-4": `
+    inset 0 -.5px 0 0 hsl(var(--shadow-color) / calc(var(--shadow-strength) + 9%))`,
+};
```
Comment (argyleink) [inline src/props.shadows.js:52]: looks like another typo here because the inner shadows are now *only* the highlight and don't have the inner shadow part

Screenshot 2023-08-29 at 3 41 37 PM

Comment (argyleink) [reply src/props.shadows.js:52]: just to be clear too, this PR should really only be changing these lines. need to combine the highlights like you added above, with the previous shadows that were there. thinking about it too, the highlight could be another custom property, which would help us not repeat the same highlight shadow over and over.

**Review by argyleink — CHANGES_REQUESTED** [review]

**Comment by argyleink** [comment]
here! i wont merge this because i want you to get contribution credit, but this is the PR shape I'm lookin for 🙂 

https://github.com/argyleink/open-props/pull/399

**Comment by argyleink** [comment]
hi @Vidhanvyrs, you still interested in contributing this? #399 is ready to go, shows all the changes needing to be made, but I'd still totally prefer if you got credit for the work you've done so far so I don't want to merge it 🙂

**Comment by Vidhanvyrs** [comment]
No thankyou its fine you lot can continue BTW thanx!!🙇‍♂️
